docs/python_support/reference.md

```diff
@@ -0,0 +1,769 @@
+# Python Nodes in OpenVINO Model Server {#ovms_docs_python_support_reference}
+
+## Introduction
+
+**This feature is currently in preview, meaning some behaviors of the feature and user interface may change in future versions**
+
+ Starting with version 2023.3, OpenVINO Model Server supports execution of custom Python code. Such code can execute simple pre- or post-processing as well as complex tasks like image or text generation. 
+ 
+ Python execution is enabled via [MediaPipe](../mediapipe.md) by the built-in `PythonExecutorCalculator` that allows creating graph nodes to execute Python code. Python nodes can be used as standalone servables (single node graphs) or be part of larger MediaPipe graphs.
+
+ Check out the [quickstart guide](quickstart.md) for a simple example that shows how to use this feature.
+
+ ## Building Docker Image
+
+The publicly available `openvino/model_server` image on Docker Hub supports Python, but does not come with external modules installed. If Python is all you need then you can use the public image without modification. Otherwise, you will need to extend the public image with additional layers that install any modules required for your Python code to run. For example, let's say your code requires numpy.
+In that case, your Dockerfile may look like this:
+
+```dockerfile
+FROM openvino/model_server:latest
+USER root
+ENV LD_LIBRARY_PATH=/ovms/lib
+ENV PYTHONPATH=/ovms/lib/python
+RUN apt update && apt install -y python3-pip git
+RUN pip3 install numpy
+ENTRYPOINT [ `/ovms/bin/ovms` ]
+```
+
+You can also modify `requirements.txt` from our [python demos](../../demos/python_demos) and from repository top level directory run `make python_image` 
+
+## `OvmsPythonModel` class
+
+When deploying a Python node, the Model Server expects a Python file with an `OvmsPythonModel` class implemented:
+
+```python
+class OvmsPythonModel:
+
+    def initialize(self, kwargs):
+        """
+        `initialize` is called when model server loads graph definition. 
+        It allows to initialize and maintain state between subsequent execute() calls and even graph instances. 
+        For gRPC unary, graphs are recreated per request. 
+        For gRPC streaming, there can be multiple graph instances existing at the same time. 
+        OvmsPythonModel object is initialized with this method and then shared between all graph instances. 
+        Implementing this function is optional.
+
+        Parameters:
+        -----------
+        kwargs : dict
+            Available arguments:
+            * node_name: string
+                Name of the node in the graph
+            * input_streams: list of strings
+                List of input stream names defined for the node in graph
```

Comment:
It is not clear if it contains "TAG:name" or just "name". I think previously it was containing tag, is it still like that?

Reply:
Ok, now I see it explained in dept below